### PR TITLE
Add rook cluster chart - WIP

### PR DESCRIPTION
**Pull request by pierreozoux**
This is work in progress.

Question is do you have any ideas you wanted to see in this `cluster` chart?

### CODE REVIEW

**Comment by pierreozoux** [comment]
Ok, I did some tests, incorporated the ideas of #710 and I managed to create a working cluster.

<details>
  <summary>Output of the helm debug</summary>

```
$ helm install --dry-run --debug --name rook --namespace rook ./
[debug] Created tunnel using local port: '42762'

[debug] SERVER: "127.0.0.1:42762"

[debug] Original chart version: ""
[debug] CHART PATH: /root/rook/cluster/charts/rook-cluster

NAME:   rook
REVISION: 1
RELEASED: Wed Mar  7 13:42:02 2018
CHART: rook-cluster-0.0.1
USER-SUPPLIED VALUES:
{}

COMPUTED VALUES:
cluster:
  backend: ceph
  dataDirHostPath: /var/lib/rook
  hostNetwork: false
  monCount: 3
  storage:
    storeConfig:
      databaseSizeMB: 1024
      journalSizeMB: 1024
      storeType: bluestore
    useAllDevices: false
    useAllNodes: true
pools:
- name: replicapool
  spec:
    replicated:
      size: 1
  storageClass:
    default: false
    enabled: true

HOOKS:
MANIFEST:

---
# Source: rook-cluster/templates/configmap.yaml
kind: ConfigMap
apiVersion: v1
metadata:
  name: rook-config-override
  labels:
    rook_cluster: rook
    chart: "rook-cluster-0.0.1"
data:
  config: ""
---
# Source: rook-cluster/templates/storageclass.yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
    name: rook-replicapool-block
    label:
      chart: "rook-cluster-0.0.1"
      rook_cluster: rook
provisioner: rook.io/block
parameters:
  pool: replicapool
  clusterName: rook
---
# Source: rook-cluster/templates/cluster.yaml
apiVersion: rook.io/v1alpha1
kind: Cluster
metadata:
  name: rook
  namespace: rook
  label:
    rook_cluster: rook
spec:
  backend: ceph
  dataDirHostPath: /var/lib/rook
  hostNetwork: false
  monCount: 3
  storage:
    storeConfig:
      databaseSizeMB: 1024
      journalSizeMB: 1024
      storeType: bluestore
    useAllDevices: false
    useAllNodes: true
---
# Source: rook-cluster/templates/pool.yaml
apiVersion: rook.io/v1alpha1
kind: Pool
metadata:
  name: replicapool
  namespace: rook
  label:
    chart: "rook-cluster-0.0.1"
    rook_cluster: rook
spec:
  replicated:
    size: 1
```
</details>

I didn't add the other objects yet, but I think it is good enough to get started.

File: cluster/charts/rook-cluster/templates/NOTES.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+The Rook Cluster has been installed.
```
Comment (hunter) [inline cluster/charts/rook-cluster/templates/NOTES.txt]: Can we add some more detail here about how to use the cluster after installation?

File: cluster/charts/rook-cluster/README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+To uninstall/delete the `my-release` deployment:
+
+```bash
+$ helm delete my-release
+```
+
+The command removes all the Kubernetes components associated with the chart and deletes the release.
+
+## Configuration
```
Comment (hunter) [inline cluster/charts/rook-cluster/README.md]: The sections below seem to be from a MySQL chart?

**Comment by hunter** [comment]
My previous PR had support for Object Storage Service/Ingress as back then there was no Object Storage CRD. Should support be added for the new Object and FileSystem CRDs?

**Comment by pierreozoux** [comment]
I'll actually recreate a pull request with this patten:
https://github.com/kubernetes/charts/pull/3795

**Pull request closed without merging**